js/misc/paygManager.js

```diff
@@ -60,33 +60,64 @@ const DBusErrorsMapping = {
     DISABLED          : 'com.endlessm.Payg1.Error.Disabled',
 }
 
+// This function checks the configuration file of PAYG directly
+// from the expected locations on disk, on an attempt to figure
+// out whether the feature is enabled, so that we don't wake up
+// the D-Bus service and keep it running when it's disabled.
+function _isPaygEnabled() {
+    // See man page eos-payg.conf(5)
+    let searchDirs = [
+        '/etc/eos-payg',
+        '/usr/local/share/eos-payg',
+        '/usr/share/eos-payg',
+    ];
+
+    let configFileName = 'eos-payg.conf'
+    let keyfile = new GLib.KeyFile();
+    try {
+        keyfile.load_from_dirs(configFileName,
+                               searchDirs,
+                               GLib.KeyFileFlags.NONE);
+        return keyfile.get_boolean('PAYG', 'Enabled');
+    } catch (e) {
+        // A non-existent file is a perfectly normal use case.
+        if (!e.matches(GLib.KeyFileError, GLib.KeyFileError.NOT_FOUND))
+            logError(e, "Error reading PAYG configuration file from " + configFileName);
+    }
+
+    return false;
+}
+
 var PaygManager = new Lang.Class({
     Name: 'PaygManager',
 
     _init: function() {
-        this._initialized = false;
-
-        this._proxy = null;
-        this._proxyInfo = Gio.DBusInterfaceInfo.new_for_xml(EOS_PAYG_IFACE);
-
         this._enabled = false;
         this._expiryTime = 0;
         this._rateLimitEndTime = 0;
 
-        this._codeExpiredId = 0;
-        this._propertiesChangedId = 0;
+        // Consider this manager initialized if PAYG is not
+        // enabled, and skip all the D-Bus related bits.
+        this._initialized = !_isPaygEnabled();
+        if (!this._initialized) {
+            this._proxy = null;
```

Comment:
I think you always want to explicitly initialize this._proxy to null, even if payg is not enabled.

Reply:
Not necessarily. The only two places where the proxy is needed (addCode() and clearCode()) are protected with `if (!this._proxy)`, which will be `false` if `this._proxy == undefined`.